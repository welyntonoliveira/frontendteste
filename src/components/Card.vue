<template>
    <div class="card">
        <div class="container_avatar-funcionario">
            <img :src="avatar" alt="Avatar do funcionário" />
            <div class="info-funcionarios_status">
                <span :class="[status ? 'bandeira_ativo' : 'bandeira_inativo', 'bandeira']">{{status? "ativo" : "inativo"}}</span>
            </div>
        </div>
        <div :class="[vermais ? 'info_home' : 'info_detalhes', 'container_info-funcionario']">
            <div class="info-funcionario_nome">
                <h3>{{ nome }}</h3>
            </div>
            <div class="info-funcionario_cod">
                <p>Código: {{ code }}</p>
            </div>
            <div class="info_acao-funcionario">
                <button v-if="vermais" @click="navigateToDetail">Ver mais</button>
            </div>
        </div>
    </div>
</template>

<script setup lang="ts">
import { defineProps } from 'vue';
import { useRouter } from 'vue-router';

// Definindo as props do componente
const props = defineProps<{
    avatar: string,
    nome: string,
    code: string,
    id: [String, Number],
    status: boolean,
    vermais: boolean,
}>();

const router = useRouter();

const navigateToDetail = () => {
    router.push({ name: 'funcionarioDetalhe', params: { id: props.id } });
};

</script>


<style scoped>
.card {
    background-color: #fff;
    border: 1px solid rgba(33,33,33,0.3);
    border-radius: 4px;
    padding: 16px;
    -webkit-box-shadow: 0px 0px 1px 0px rgba(33,33,33,0.3);
    -moz-box-shadow: 0px 0px 1px 0px rgba(33,33,33,0.3);
    box-shadow: 0px 0px 1px 0px rgba(33,33,33,0.3);
    display: flex;
    flex-flow: column;
    gap: 16px;
    justify-content: center;
    align-items: center;
    min-width: 250px;
}

.container_info-funcionario {
    width: 100%;
    text-align: center;
}

.container_avatar-funcionario img {
    width: 100px;
    height: 100px;
    border-radius: 50%;
    object-fit: none;
}

.info-funcionario_nome h3{
    font-weight: 700;
}

.info-funcionarios_status {
    margin: 4px 0 0;
    text-align: center;
}

.bandeira {
    font-weight: 700;
    line-height: 1.2em;
    letter-spacing: 1px;
}

.bandeira_ativo {
    color: #2E9629;
}

.bandeira_inativo {
    color: #B83235;
}

.info_acao-funcionario {
    text-align: center;
    margin: 24px 0 0;
}

.info_detalhes {
    text-align: left;
}

@media (max-width: 600px) {
    .card {
    min-width: 100%;
  }
}
</style>
