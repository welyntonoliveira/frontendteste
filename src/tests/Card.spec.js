import { mount } from '@vue/test-utils';
import Card from '@components/Card.vue'; 
import { createRouter, createWebHistory } from 'vue-router';

describe('Card', () => {
  const router = createRouter({
    history: createWebHistory(),
    routes: [
      { path: '/funcionario/:id', name: 'funcionarioDetalhe', component: { template: '<div>Detalhe do Funcionário</div>' } }
    ]
  });

  const mountComponent = (props) => {
    return mount(Card, {
      global: {
        plugins: [router]
      },
      props
    });
  };

  it('renders correctly with given props', () => {
    const wrapper = mountComponent({
      avatar: 'http://example.com/avatar.jpg',
      nome: 'John Doe',
      code: '1234',
      id: '1',
      status: true,
      vermais: true
    });

    expect(wrapper.find('img').attributes('src')).toBe('http://example.com/avatar.jpg');
    expect(wrapper.find('h3').text()).toBe('John Doe');
    expect(wrapper.find('.info-funcionario_cod p').text()).toBe('Código: 1234');
    expect(wrapper.find('.bandeira').text()).toBe('ativo');
    expect(wrapper.find('.bandeira').classes()).toContain('bandeira_ativo');
  });

  it('renders "Ver mais" button when vermais is true', () => {
    const wrapper = mountComponent({
      avatar: 'http://example.com/avatar.jpg',
      nome: 'John Doe',
      code: '1234',
      id: '1',
      status: true,
      vermais: true
    });

    expect(wrapper.find('button').exists()).toBe(true);
    expect(wrapper.find('button').text()).toBe('Ver mais');
  });

  it('does not render "Ver mais" button when vermais is false', () => {
    const wrapper = mountComponent({
      avatar: 'http://example.com/avatar.jpg',
      nome: 'John Doe',
      code: '1234',
      id: '1',
      status: true,
      vermais: false
    });

    expect(wrapper.find('button').exists()).toBe(false);
  });

  it('displays the correct status class based on the status prop', () => {
    const wrapper = mountComponent({
      avatar: 'http://example.com/avatar.jpg',
      nome: 'John Doe',
      code: '1234',
      id: '1',
      status: false,
      vermais: true
    });

    expect(wrapper.find('.bandeira').text()).toBe('inativo');
    expect(wrapper.find('.bandeira').classes()).toContain('bandeira_inativo');
  });
});
