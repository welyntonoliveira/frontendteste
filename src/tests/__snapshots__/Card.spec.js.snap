// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly 1`] = `
"<div class="card">
  <div class="container_avatar-funcionario"><img src="path/to/avatar.png" alt="Avatar do funcion치rio">
    <div class="info-funcionarios_status"><span class="bandeira_ativo bandeira">ativo</span></div>
  </div>
  <div class="info_home container_info-funcionario">
    <div class="info-funcionario_nome">
      <h3>John Doe</h3>
    </div>
    <div class="info-funcionario_cod">
      <p>C칩digo: 12345</p>
    </div>
    <div class="info_acao-funcionario"><button>Ver mais</button></div>
  </div>
</div>"
`;

exports[`Card.vue renders correctly with basic props 1`] = `
"<div class="card">
  <div class="container_avatar-funcionario"><img src="https://example.com/avatar.jpg" alt="Avatar do funcion치rio">
    <div class="info-funcionarios_status"><span class="bandeira_ativo bandeira">ativo</span></div>
  </div>
  <div class="info_detalhes container_info-funcionario">
    <div class="info-funcionario_nome">
      <h3>Fulano da Silva</h3>
    </div>
    <div class="info-funcionario_cod">
      <p>C칩digo: 12345</p>
    </div>
    <div class="info_acao-funcionario">
      <!--v-if-->
    </div>
  </div>
</div>"
`;
